### PR TITLE
feat: Syncing read status now updates the read history on MangaDex

### PR DESCRIPTION
**Pull request by sassy-lily**
## Summary
- Adds the optional `?updateHistory=true` query parameter when POSTing read markers to `/manga/{id}/read`, so chapters read in the app are recorded in the user's MangaDex reading history feed in addition to the binary read-marker set.
- Live, user-driven read events opt in to `updateHistory=true`: reader auto-mark on completion, mark / mark-previous-as-read from the manga screen, and the download-notification "Mark as read" action. Unread paths and any future bulk-backfill stay at the handler default of `false`.
- Plumbing change (Retrofit method + `StatusHandler.markChaptersStatus` signature) is split from the behaviour change so each commit is reviewable independently.

## Test plan
- [ ] Sign into MangaDex with `Settings → MangaDex → Reading sync` enabled.
- [ ] Read a chapter to completion in the reader; confirm it appears at https://mangadex.org/user/me/history.
- [ ] On a manga screen, "Mark previous as read" on a few chapters; confirm those entries appear in the history feed.
- [ ] Mark a chapter as unread; confirm the request still succeeds and does not generate a history entry.

## Out of scope
- The download-notification "Mark as read" path also has a separate bug (passes the local Room PK as the manga UUID, so the request 4xxs silently). The `updateHistory=true` flag is wired in this PR ahead of that fix; the path itself will be addressed in a follow-up.
- `ChapterSourceSync` calls `markMergedChaptersStatus` with non-merged chapters, which silently no-ops. Also a follow-up.

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt [hunk 1]
```diff
@@ -46,14 +46,15 @@ class StatusHandler {
         mangaId: String,
         chapterIds: List<String>,
         read: Boolean = true,
+        updateHistory: Boolean = false,
```
Comment (nonproto) [inline app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt:49]: Should default this to true

File: app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderViewModel.kt [hunk 2]
```diff
@@ -1099,6 +1099,7 @@ constructor(
                 statusHandler.markChaptersStatus(
                     manga!!.uuid(),
                     listOf(readerChapter.chapter.mangadex_chapter_id),
+                    updateHistory = true,
```
Comment (nonproto) [inline app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderViewModel.kt:1102]: don't need to pass this in now

File: app/src/main/java/org/nekomanga/usecases/chapters/MarkChaptersRemote.kt [hunk 3]
```diff
@@ -36,6 +36,7 @@ class MarkChaptersRemote(
                         mangaUuid,
                         nonMergedChapters.map { it.mangaDexChapterId },
                         syncRead,
+                        updateHistory = syncRead,
```
Comment (nonproto) [inline app/src/main/java/org/nekomanga/usecases/chapters/MarkChaptersRemote.kt:39]: This also should always be true.  If we are marking something as unread we want to update the history to be unread.  If we are marking something read we want the history to be updated

**Review by nonproto — CHANGES_REQUESTED** [review]

File: app/src/main/java/eu/kanade/tachiyomi/data/notification/NotificationReceiver.kt [hunk 4]
```diff
@@ -301,6 +301,7 @@ class NotificationReceiver : BroadcastReceiver() {
                 statusHandler.markChaptersStatus(
                     mangaId.toString(),
                     nonMergedChapters.map { it.mangadex_chapter_id },
+                    updateHistory = true,
```
Comment (nonproto) [inline app/src/main/java/eu/kanade/tachiyomi/data/notification/NotificationReceiver.kt:304]: Not needed when it defaults to true

File: app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt (hunk 1 above)
Comment (nonproto) [reply app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt:49]: Actually I don't see a scenario where this ever going to be false so should we shouldnt even expose it.  Just hardcode true

Comment (sassy-lily) [reply app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt:49]: > Actually I don't see a scenario where this ever going to be false so should we shouldnt even expose it. Just hardcode true

It's mainly for backwards-compatibility, so callers must explicitly subscribe to the new behavior.

If you say it's fine to change the behavior for every caller at the same time then yeah it's not needed.

Comment (sassy-lily) [reply app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt:49]: > Actually I don't see a scenario where this ever going to be false so should we shouldnt even expose it. Just hardcode true

Doing so has a side-effect: when a chapter is marked as unread in the app, the change will be passed to MangaDex with the `updateHistory` (since it will be hardcoded) and the user will find that chapter in its history.

Let me know if this is acceptable so I can apply the changes.

File: app/src/main/java/org/nekomanga/usecases/chapters/MarkChaptersRemote.kt (hunk 3 above)
Comment (sassy-lily) [reply app/src/main/java/org/nekomanga/usecases/chapters/MarkChaptersRemote.kt:39]: > If we are marking something as unread we want to update the history to be unread.

This will have the chapter appear in the user's *read* list, though (unless MangaDex performs something on their side).

It's not completely appropriate (although harmless).

File: app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt (hunk 1 above)
Comment (nonproto) [reply app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt:49]: hmm I assumed marking unread would remove it from read history on Dex.

Comment (sassy-lily) [reply app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt:49]: The documentation is unclear about it:

https://api.mangadex.org/docs/redoc.html#tag/ReadMarker/operation/post-manga-chapter-readmarkers

> updateHistory
> Adding this will cause the chapter to be stored in the user's reading history

It doesn't mention the edge case where you both set the parameter and provide only `chapterIdsUnread` entries.

Short of a live test I think there's no way of knowing.

Comment (nonproto) [reply app/src/main/java/eu/kanade/tachiyomi/source/online/handlers/StatusHandler.kt:49]: I am asking their dev api channel

**Comment by nonproto** [comment]
I think this can be closed.  The UI on MangaDex mentions:

`Reading history is currently only tracked on your device, and will be lost if you log out or clear site data.`

So this flag is going to do nothing
